addons/survey/controllers/survey_session_manage.py

```diff
@@ -27,7 +27,6 @@ def _fetch_from_session_code(self, session_code):
         if session_code:
             matching_survey = request.env['survey.survey'].sudo().search([
                 ('state', '=', 'open'),
-                ('session_state', 'in', ['ready', 'in_progress']),
```

Comment:
Please update docstring accordingly as we rae still talking about "We force the session_state to be reachable (ready / in_progress) to avoid people using this route to access other (private) surveys." Don't just delete the line, add some explanations :D